aten/src/THC/THCTensorRandom.cu

```diff
@@ -47,35 +31,24 @@ THC_API __host__ void THCRandom_getRNGState(THCState* state, THByteTensor *rng_s
   THCGenerator* gen = THCRandom_getGenerator(state);
   std::lock_guard<std::mutex> lock(gen->mutex);
 
-  // The RNG state comprises the MTPG32 states, the seed, and an offset used for Philox
-  static const size_t states_size = MAX_NUM_BLOCKS * sizeof(curandStateMtgp32);
+  // The RNG state comprises the seed, and an offset used for Philox
+  static const size_t states_size = MAX_NUM_BLOCKS * sizeof(curandStateMtgp32); // this line is just here for BC reason
   static const size_t seed_size = sizeof(gen->state.initial_seed);
   static const size_t offset_size = sizeof(gen->state.philox_seed_offset);
   static const size_t total_size = states_size + seed_size + offset_size;
   THByteTensor_resize1d(rng_state, total_size);
   THArgCheck(THByteTensor_nElement(rng_state) == total_size, 1, "RNG state is wrong size");
   THArgCheck(THByteTensor_isContiguous(rng_state), 1, "RNG state must be contiguous");
-  THCudaCheck(cudaMemcpy(THByteTensor_data(rng_state), gen->state.gen_states,
-                         states_size, cudaMemcpyDeviceToHost));
```

Comment:
It might be a good idea to fill in this memory with deterministic garbage so if someone tries to use it (improperly) it won't be a random error

Reply:
Filled in the memory with -1 and verified locally that torch.cuda.get_rng_state() gives 255 in the first few elements.